public/app/features/explore/state/query.ts

```diff
@@ -696,6 +695,103 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
   }
 );
 
+interface RunLoadMoreLogsQueriesOptions {
+  exploreId: string;
+  absoluteRange: AbsoluteTimeRange;
+}
+/**
+ * Supplementary action to run queries that request more results, and dispatches sub-actions based
+ * on which result viewers are active
+ */
+export const runLoadMoreLogsQueries = createAsyncThunk<void, RunLoadMoreLogsQueriesOptions>(
+  'explore/runQueries',
+  async ({ exploreId, absoluteRange }, { dispatch, getState }) => {
+    dispatch(cancelQueries(exploreId));
+
+    const { datasourceInstance, containerWidth, queryResponse } = getState().explore.panes[exploreId]!;
+    const { defaultCorrelationEditorDatasource, scopedVars, showCorrelationEditorLinks } = await getCorrelationsData(
+      getState(),
+      exploreId
+    );
+    const correlations$ = getCorrelations(exploreId);
+
+    let newQuerySource: Observable<ExplorePanelData>;
+
+    const logQueries = queryResponse.logsResult?.queries || [];
+    const queries = logQueries.map((query: DataQuery) => ({
+      ...query,
+      datasource: query.datasource || datasourceInstance?.getRef(),
+    }));
+
+    if (!hasNonEmptyQuery(queries) || !datasourceInstance) {
+      return;
+    }
+
+    // Some datasource's query builders allow per-query interval limits,
+    // but we're using the datasource interval limit for now
+    const minInterval = datasourceInstance?.interval;
+
+    const queryOptions: QueryOptions = {
+      minInterval,
+      // maxDataPoints is used in:
+      // Loki - used for logs streaming for buffer size, with undefined it falls back to datasource config if it supports that.
+      // Elastic - limits the number of datapoints for the counts query and for logs it has hardcoded limit.
+      // Influx - used to correctly display logs in graph
+      // TODO:unification
```

Comment:
Should this be addressed in this PR?

Reply:
Same as before: https://github.com/grafana/grafana/blob/master/public/app/features/explore/state/query.ts#L594

Should I maybe remove these comments?

Reply:
Yep, I think we can remove it. Was added in https://github.com/grafana/grafana/pull/25890 and looks obsolete to me.

Reply:
Thanks, I agree.